mmcls/models/losses/label_smooth_loss.py

```diff
@@ -1,50 +1,69 @@
-import torch
-import torch.nn as nn
-import torch.nn.functional as F
+import numpy as np
 
 from ..builder import LOSSES
-from .utils import weight_reduce_loss
+from .cross_entropy_loss import CrossEntropyLoss
+from .utils import convert_to_one_hot
 
 
-def label_smooth(pred,
-                 label,
-                 label_smooth_val,
-                 avg_smooth_val,
-                 weight=None,
-                 reduction='mean',
-                 avg_factor=None):
-    # # element-wise losses
-    one_hot = torch.zeros_like(pred)
-    one_hot.fill_(avg_smooth_val)
-    label = label.view(-1, 1)
-    one_hot.scatter_(1, label, 1 - label_smooth_val + avg_smooth_val)
-
-    loss = -torch.sum(F.log_softmax(pred, 1) * (one_hot.detach()))
-
-    # apply weights and do the reduction
-    if weight is not None:
-        weight = weight.float()
-    loss = weight_reduce_loss(
-        loss, weight=weight, reduction=reduction, avg_factor=avg_factor)
+@LOSSES.register_module()
+class LabelSmoothLoss(CrossEntropyLoss):
+    """Intializer for the label smoothed cross entropy loss.
 
-    return loss
+    This decreases gap between output scores and encourages generalization.
+    Labels provided to forward can be one-hot like vectors (NxC) or class
+    indices (Nx1).
+    This normalizes the labels to a sum of 1 based on the total count of
+    positive targets for a given sample before applying label smoothing.
 
-
-@LOSSES.register_module()
-class LabelSmoothLoss(nn.Module):
+    Args:
+        label_smooth_val (float): value to be added to each target entry
+        num_classes (int): the number of classes. Defaults to None.
```

Comment:
as usual, if the defaulted value is None, the argument is optional.  
like....num_classes (int, optional): the number of classes. Defaults to None.